### PR TITLE
Skip visdiff on protocol-only-changes

### PR DESCRIPTION
**Pull request by chrisnojima**
Don't run visdiff if only protocol changes happen

- [x] remove protocol changes

### CODE REVIEW

**Comment by chrisnojima** [comment]
cc: @maxtaco

**Comment by maxtaco** [comment]
thanks!

**Comment by maxtaco** [comment]
BTW, might need a rebase to make this pass CI. LMK if that doesn't work

File: shared/jenkins_test.sh [hunk 1]
```diff
@@ -26,17 +26,19 @@ check_rc() {
...
 
 has_js_files() {
+    extra_commands=$1
     if [ $against_master -eq 1 ]; then
         echo 'Missing $change_target, forcing has_js_files to true'
         return
     fi
     echo 'git fetch'
-    git fetch
+    # git fetch
```
Comment (chromakode) [inline shared/jenkins_test.sh]: Why comment this out?

Comment (chrisnojima) [inline shared/jenkins_test.sh]: leftover from testing, put back tnx!

**Comment by chrisnojima** [comment]
- [ ] ensure visdiff checks but doesn't run

**Comment by chrisnojima** [comment]
- [ ] visdiff not working, need to see if argv is messed up maybe
```17:21:48 [shared] Running shell script
17:21:49 + node ./jenkins-test.js js 6522edd20afc8cf5a17ac00a14189bed1af46247 master
17:21:49 fatal: Not a valid object name /home/jenkins/.nvm/versions/node/v6.9.2/bin/node,/tmp/jenkins/workspace/client/PR-7367/5/go/src/github.com/keybase/client/shared/jenkins-test.js,js,6522edd20afc8cf5a17ac00a14189bed1af46247,masterorigin/master
17:21:49 child_process.js:526
17:21:49     throw err;
17:21:49     ^
17:21:49 
17:21:49 Error: Command failed: git merge-base /home/jenkins/.nvm/versions/node/v6.9.2/bin/node,/tmp/jenkins/workspace/client/PR-7367/5/go/src/github.com/keybase/client/shared/jenkins-test.js,js,6522edd20afc8cf5a17ac00a14189bed1af46247,masterorigin/master 6522edd20afc8cf5a17ac00a14189bed1af46247
17:21:49 fatal: Not a valid object name /home/jenkins/.nvm/versions/node/v6.9.2/bin/node,/tmp/jenkins/workspace/client/PR-7367/5/go/src/github.com/keybase/client/shared/jenkins-test.js,js,6522edd20afc8cf5a17ac00a14189bed1af46247,masterorigin/master
17:21:49 ```

**Comment by keybase-ci-visdiff** [comment]
The commits b1fafd18baec...0608e17b0c04 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498670631794-b1fafd18baec-0608e17b0c04-linux/Markdown-emoji-b1fafd18baec.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498670631794-b1fafd18baec-0608e17b0c04-linux/Markdown-emoji-0608e17b0c04.png)  
   Markdown-emoji comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits b1fafd18baec...01131d321f97 introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498675901720-b1fafd18baec-01131d321f97-linux/Tracker-New_user_Scroll_1-b1fafd18baec.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498675901720-b1fafd18baec-01131d321f97-linux/Tracker-New_user_Scroll_1-01131d321f97.png)  
   Tracker-New_user_Scroll_1 comparison
 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498675901720-b1fafd18baec-01131d321f97-linux/Markdown-emoji-b1fafd18baec.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498675901720-b1fafd18baec-01131d321f97-linux/Markdown-emoji-01131d321f97.png)  
   Markdown-emoji comparison
</summary>

**Comment by chrisnojima** [comment]
- [ ] waiting on visdiff to show up on ui changes
- [ ] cleanup temp code

**Comment by keybase-ci-visdiff** [comment]
The commits 9c012fcf9d8b...4bfd568ec71d introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **Chat_Conversations_List-Selected_Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498876727701-9c012fcf9d8b-4bfd568ec71d-linux/Chat_Conversations_List-Selected_Normal-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498876727701-9c012fcf9d8b-4bfd568ec71d-linux/Chat_Conversations_List-Selected_Normal-4bfd568ec71d.png)  
   Chat_Conversations_List-Selected_Normal comparison
 * Changed: **Chat_Conversations_List-Selected_Muted** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498876727701-9c012fcf9d8b-4bfd568ec71d-linux/Chat_Conversations_List-Selected_Muted-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498876727701-9c012fcf9d8b-4bfd568ec71d-linux/Chat_Conversations_List-Selected_Muted-4bfd568ec71d.png)  
   Chat_Conversations_List-Selected_Muted comparison
 * Changed: **Chat_Conversations_List-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498876727701-9c012fcf9d8b-4bfd568ec71d-linux/Chat_Conversations_List-Normal-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498876727701-9c012fcf9d8b-4bfd568ec71d-linux/Chat_Conversations_List-Normal-4bfd568ec71d.png)  
   Chat_Conversations_List-Normal comparison
</summary>

**Comment by chrisnojima** [comment]
nearly 20 days later this thing finally seems like its working...
@chromakode cause you looked at this awhile ago

**Comment by chrisnojima** [comment]
and of course it fails again, sigh

**Comment by keybase-ci-visdiff** [comment]
The commits 9c012fcf9d8b...59b54bc1a2a8 introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **Chat_Conversations_List-Selected_Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498877699408-9c012fcf9d8b-59b54bc1a2a8-linux/Chat_Conversations_List-Selected_Normal-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498877699408-9c012fcf9d8b-59b54bc1a2a8-linux/Chat_Conversations_List-Selected_Normal-59b54bc1a2a8.png)  
   Chat_Conversations_List-Selected_Normal comparison
 * Changed: **Chat_Conversations_List-Selected_Muted** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498877699408-9c012fcf9d8b-59b54bc1a2a8-linux/Chat_Conversations_List-Selected_Muted-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498877699408-9c012fcf9d8b-59b54bc1a2a8-linux/Chat_Conversations_List-Selected_Muted-59b54bc1a2a8.png)  
   Chat_Conversations_List-Selected_Muted comparison
 * Changed: **Chat_Conversations_List-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498877699408-9c012fcf9d8b-59b54bc1a2a8-linux/Chat_Conversations_List-Normal-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498877699408-9c012fcf9d8b-59b54bc1a2a8-linux/Chat_Conversations_List-Normal-59b54bc1a2a8.png)  
   Chat_Conversations_List-Normal comparison
</summary>

**Comment by chrisnojima** [comment]
ok this is working again
- [ ] try protocl only change
- [ ] try gui change

**Comment by keybase-ci-visdiff** [comment]
The commits 9c012fcf9d8b...76149ae8be73 introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **Chat_Conversations_List-Selected_Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499367818580-9c012fcf9d8b-76149ae8be73-linux/Chat_Conversations_List-Selected_Normal-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499367818580-9c012fcf9d8b-76149ae8be73-linux/Chat_Conversations_List-Selected_Normal-76149ae8be73.png)  
   Chat_Conversations_List-Selected_Normal comparison
 * Changed: **Chat_Conversations_List-Selected_Muted** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499367818580-9c012fcf9d8b-76149ae8be73-linux/Chat_Conversations_List-Selected_Muted-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499367818580-9c012fcf9d8b-76149ae8be73-linux/Chat_Conversations_List-Selected_Muted-76149ae8be73.png)  
   Chat_Conversations_List-Selected_Muted comparison
 * Changed: **Chat_Conversations_List-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499367818580-9c012fcf9d8b-76149ae8be73-linux/Chat_Conversations_List-Normal-9c012fcf9d8b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499367818580-9c012fcf9d8b-76149ae8be73-linux/Chat_Conversations_List-Normal-76149ae8be73.png)  
   Chat_Conversations_List-Normal comparison
</summary>

**Comment by chrisnojima** [comment]
one last sanity check:

- [x] protocol only change no visdiff
- [ ] visdiff change does visdiff

**Comment by chrisnojima** [comment]
i think this works again

**Comment by chrisnojima** [comment]
@chromakode ok this pr is working again. if you have just a proto change it'll skip visdiff, otherwise it'll run

**Comment by chrisnojima** [comment]
@cjb

**Comment by chrisnojima** [comment]
@MarcoPolo @cjb this skipps visdiff runs if you only make protocol changes

**Pull request merged**